### DISCUSSION TITLE
Labels of agents with lower layer level don't show up in the visualization webpage

### DISCUSSION THREAD

**Discussion by FoFFolo** [mesa/mesa#2056] [Q&A] [answered]
Hello, I have a problem with the webpage of my ABM. It seems that when an agent changes its layer to a lower one(the only layers level are 0 and 1), when I hold the pointer to that agent, the label showing its text and layer doesn't show up.
The problem occurs in the portrayal function when it modifies the layer level of each agent based on his wealth:
![Screenshot 2024-02-26 132108](https://github.com/projectmesa/mesa/assets/74419371/08dcd875-9e7b-4e59-8951-9bade3abfb2e)

![Screenshot 2024-02-26 132238](https://github.com/projectmesa/mesa/assets/74419371/345fae13-f45e-47cc-b89c-d2a42a724aca)
At the start point, every agent has the same layer level, so the label shows up on each of them, but when an agent becomes red, then its layer becomes 1, and after that only the agents with layer 1 will have their label showing up, while the green agents with layer 0 will not.
![Screenshot 2024-02-26 132445](https://github.com/projectmesa/mesa/assets/74419371/16ee63ad-4ebc-4034-944d-5dd6fdde4cd3)
![Screenshot 2024-02-26 133350](https://github.com/projectmesa/mesa/assets/74419371/f7c668ff-b85c-4383-9b19-072937962a5b)

I tried to restart the server, or change some values but nothing has changed.
I attach the zip file of the project, it's small and it only requires to run the run.py script in order to run the server.
[starter_model_with_server.zip](https://github.com/projectmesa/mesa/files/14404672/starter_model_with_server.zip)

**Comment by rht** [comment, marked as answer]
Sorry for the late response. Yes, it is the limitation of the current mesa-viz-tornado. I personally am not actively improving mesa-viz-tornado anymore. But label tooltip should be doable with the [Altair backend](https://github.com/projectmesa/mesa/blob/main/mesa/experimental/components/altair.py) of the new experimental visualization. It is also not implemented yet, but it is easier to do with the new pure-Python experimental visualization, with just 1 or 2 lines, e.g. https://altair-viz.github.io/gallery/scatter_tooltips.html.
